tests/unit/test_client.py

```diff
@@ -136,17 +156,30 @@ def submit(self, fn, *args, **kwargs):
 
 class TestAzStorageTorchBlobClient:
     def assert_expected_download_calls(
-        self, mock_generated_sdk_storage_client, expected_ranges, expected_etag
+        self,
+        mock_generated_sdk_storage_client,
+        expected_ranges,
+        expected_etag,
+        known_blob_size=False,
     ):
-        expected_download_calls = [
-            mock.call(
-                range=f"bytes={expected_range}",
-                modified_access_conditions=ModifiedAccessConditions(
-                    if_match=expected_etag
-                ),
+        expected_download_calls = []
+        start = 0
+        if not known_blob_size:
+            expected_download_calls.append(
+                mock.call(
+                    range=f"bytes={expected_ranges[0]}",
+                )
+            )
+            start = 1
+        for i in range(start, len(expected_ranges)):
```

Comment:
To simplify this down, we can use [`enumerate()`](https://docs.python.org/3/library/functions.html#enumerate) in the for loop and check for `i == 0 and not known_blob_size` to whether we inject the modified access condition as part of the expected call